pandas/util/testing.py

```diff
@@ -599,6 +605,8 @@ def _get_ilevel_values(index, level):
                                check_less_precise=check_less_precise,
                                check_exact=check_exact, obj=lobj)
             # get_level_values may change dtype
+            left = cast(MultiIndex, left)
```

Comment:
Can you move these to line 597? That is specifically where the inference of a new type should occur would logically fit better there